### ISSUE TITLE
Improve the code listing

### ISSUE THREAD

**Issue by dietmarw** [modelica-association/conference-templates#14]
I'm really unhappy the way code listing is currently handled by the template. It just looks ugly. I suggest to set the listing to use mono-spaced font which can have colour or bold face and comments can have italic in addition but that's it. Mono-spaced font funnily enough seems to be much more compact. 
I've now seen several paper making use of screenshots from codelisting (e.g., OMEdit) instead of entering the code in a listing environment and I must say that actually looks better.

**Comment by casella** [comment]
I agree 100%. I really do not understand why variable-size fonts are used with additional space to make them monospaced. By the way, a lot of precious real estate is unnecessarily wasted. I tried to improve this aspect a bit by adding
```
\lstset{basewidth={0.53	em,0.45em}}
\lstset{basicstyle= \footnotesize}
```
to the .text source, but I'm sure we could do better

**Comment by xogeny** [comment]
I also agree.  I was looking at our paper and thinking "who thinks listings should look like this?".  It has been this way in the TeX world for a long time.  Somebody, somewhere has a very warped sense of aesthetics.

**Comment by casella** [comment]
If someone can conjure up an improvement which can be confined to the modelica.cls file, we could already use it for the final submissions. It should be read in two weeks max

**Comment by akloeckner** [comment]
At DLR we were using https://code.google.com/archive/p/dtsyntax/, which renders comparably to Dymola. We also have an improved Version. I can try and integrate it tomorrow...

**Comment by dietmarw** [comment]
I think there would not be a need of changing the tool but rather tweaking the configuration. For some strange reason one of the papers using the template looks actually good by using the current setup. So please don't start introducing yet another tool just for this.

**Comment by dietmarw** [comment]
OK I found out what solves it but not why. @sjoelund might have an idea. In your paper you define an [additional listing part for c](https://github.com/modelica/Modelica_DeviceDrivers/blob/modelica2017/Modelica_DeviceDrivers/Resources/doc/Modelica2017/modelica2017_Modelica_DeviceDrivers.tex#L31) and if I paste that additional definition into the example paper the font switches (also for modelica) to monospaced. Adding the same line to the `listings-modelica.cfg` has no effect. 

So basically adding 
```
\lstset{basicstyle=\ttfamily}
```
into the `.tex` file works, but that setting is simply ignored when present ([and it is](https://github.com/modelica-association/conference-templates/blob/master/LaTeX/resources/listings-modelica.cfg#L48)) in the  `listings-modelica.cfg`. Any idea?

**Comment by dietmarw** [comment]
The nicer variant (which also takes less space is the one from the M_DD paper:
```
\lstset{basicstyle=\fontsize{9pt}{10.5pt}\ttfamily}
```

**Comment by dietmarw** [comment]
OK found the place where it was overridden. Will send a patch tomorrow. It now actually looks rather nice:
![image](https://cloud.githubusercontent.com/assets/9332/22562182/c41b5ec4-e97b-11e6-8bb2-e6bfddb226f3.png)

**Comment by dietmarw** [comment]
I'm not sure of the use of the Modelica2017 branch. @bernhard-thiele do you think we still need it?

**Comment by casella** [comment]
@dietmarw this really looks much better :)

Is it alread on master?

**Comment by dietmarw** [comment]
Yes I pushed it to master, hence me also wondering what the purpose of the modelica2017 branch should be.

**Comment by casella** [comment]
The Modelica Conference [author's guide](https://modelica.org/events/modelica2017/authors-guide/authors-guide) allows you to download the zipped file with all templates from this [url](https://github.com/modelica-association/conference-templates/archive/Modelica2017.zip). The zip file is automatically generated from the Modelica2017 branch.
Rationale:
-the templates accessible from the Conference Website should be automatically in sync with the Git repository, no need for the Chair to copy files from GIT to Clone
-in case we wanted to start developing something for the new Modelica conference (e.g. biber instead of bibtex), the Modelica2017 branch was created not to mess up things.

I would suggest that you also merge these changes to the Modelica 2017 branch, so that I can tell authors to re-download the templates and change the .cls file for the final submission.

Please confirm that only changing the modelica.cls file is necessary, no changes needed to the main .tex file.

**Comment by dietmarw** [comment]
I see. OK I've now synched the Modelica2017 branch with the latest changes from master. None of which were already changes related to the bibtex/biber setup. On the contrary, the `titlesec` fix was actually not present until just now since I wasn't aware of the function of that branch.

Anyway the only functional changes between the last version of the Modelica2017 branch and now is the [titlesec fix and the code formatting fix](https://github.com/modelica-association/conference-templates/compare/c128b75c31dc1ee092383c64b4ed0bb0ee7ccd22...master)
